common/remote/sockfile.cpp

```diff
@@ -1052,7 +1086,7 @@ class CRemoteBase: public CInterface
                         WARNLOG("Remote file authenticate %s for %s ",e->errorMessage(err).str(),ep.getUrlStr(eps.clear()).str());
                         e->Release();
                         if (!retries)
-                            break;
+                            break; // MCK - is this a warning or an error ? If an error, should we close and throw here ?
```

Comment:
Looks wrong to me.
Should fire a DAFSERR_connection_failed exception I think / shouldn't be allowed to break out of loop on failure.

This could be fixed and code cleared up, if dispensed with 'ok' flag and moved:

```
    if (AuthenticationEnabled)
        sendAuthentication(ep); // this will log error
    break;
```

after line 1056, i.e. after other connect code in main try/catch body.
The existing catch will handle failure/retries and can (if still desirable) spot that exception is a RFSERR_AuthenticateFailed and log.